compute_endpoint/globus_compute_endpoint/endpoint/endpoint_manager.py

```diff
@@ -895,32 +895,14 @@ def cmd_start_endpoint(
             os.chdir("/")  # always succeeds, so start from known place
             exit_code += 1
 
-            if (os.getuid(), os.getgid()) != (uid, gid):
+            orig_uid, orig_gid = os.getuid(), os.getgid()
+            if (orig_uid, orig_gid) != (uid, gid):
                 # For multi-user systems, this is the expected path.  But for those
                 # who run the multi-user setup as a non-privileged user, there is
                 # no need to change the user: they're already executing _as that
                 # uid_!
-                try:
-                    # The initialization of groups is "fungible" if not a
-                    # privileged user
-                    log.debug("Initializing groups for %s, %s", uname, gid)
-                    os.initgroups(uname, gid)
-                except PermissionError as e:
-                    log.warning(
-                        "Unable to initialize groups; unprivileged user?  Ignoring"
-                        " error, but further attempts to drop privileges may fail."
-                        "\n  Process ID (pid): %s"
-                        "\n  Current user: %s (uid: %s, gid: %s)"
-                        "\n  Attempted to initgroups to: %s (uid: %s, name: %s)",
-                        os.getpid(),
-                        self._mu_user.pw_name,
-                        os.getuid(),
-                        os.getgid(),
-                        gid,
-                        uid,
-                        uname,
-                    )
-                    log.debug("Exception text: (%s) %s", e.__class__.__name__, e)
+                log.debug("Initializing groups for %s, %s", uname, gid)
+                os.initgroups(uname, gid)  # raises (good!) on error
```

Comment:
Thoughts on still catching the `PermissionError`, adding a useful debug log, then re-raising?

Reply:
We do already by default.  [This entire hunk is in a `try-except`](https://github.com/globus/globus-compute/blob/2.29.0/compute_endpoint/globus_compute_endpoint/endpoint/endpoint_manager.py#L832-L1079):

```python
try:
   ...
   os.initgroups(uname, gid)
   ...
except Exception as e:
    msg = (
        f"Unable to start user endpoint process for {uname}"
        f" [exit code: {exit_code}; ({type(e).__name__}) {e}]"
    )
    log.error(msg)
    log.debug(f"Failed to exec for {uname}", exc_info=e)
    self.send_failure_notice(kwargs, msg=msg, fork=False)
...
```